lesson5/main.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+const fetchedData = new Promise((resolve, reject) => {
+  const listItems = [
+    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
+  ];
+  resolve(listItems);
+});
+
+function createNewList(data) {
+  const fragment = document.createDocumentFragment();
+  data.forEach((key) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = `/${key.to}`;
+    a.textContent = key.text;
+    img.src = key.img;
+    img.alt = key.alt;
+
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  });
+  return fragment;
+}
+
+function insertList(data) {
+  const ul = document.getElementById("js-lists");
+  ul.appendChild(data);
+}
```

Comment:
```suggestion
function insertList(fragment) {
  const ul = document.getElementById("js-lists");
  ul.appendChild(fragment);
}
```

It's fragments, right?

Reply:
Since it is predetermined in `createList` that it will return a fragment does that mean the argument here has to be a fragment?

The reason I expressed as data is

1. because this is a temporary argument, so I thought it didn't need to be that specifily.
2. If were the `createList` function to return a single li element instead of a fragment, I wanted to make it clear that it's okay if other data comes in other than fragment.

That's all.

I'm sorry if I said something wrong.

Reply:
> because this is a temporary argument, so I thought it didn't need to be that specifily.

```js
function createNewList(data) {
  const fragment = document.createDocumentFragment();
  data.forEach((key) => {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");

    a.href = `/${key.to}`;
    a.textContent = key.text;
    img.src = key.img;
    img.alt = key.alt;

    fragment
      .appendChild(li)
      .appendChild(a)
      .insertAdjacentElement("afterbegin", img);
  });
  return fragment;
}
```

this code return fragment, So function receive fragment. There is no other reason.

Reply:
I fixed code based on your advice here and on Discord.
I deconstructed insertList functions.